lib/model/utils_test.go

```diff
@@ -0,0 +1,212 @@
+// Copyright (C) 2019 The Syncthing Authors.
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+package model
+
+import (
+	"io/ioutil"
+	"os"
+	"runtime"
+	"testing"
+
+	"github.com/syncthing/syncthing/lib/fs"
+)
+
+func TestInWriteableDir(t *testing.T) {
+	dir, err := ioutil.TempDir("", "")
```

Comment:
High nitpick, ignore at your leisure: I started to put a prefix before temporary dirs such that if any of them are left behind (ctrl-c) they are easily identifiable as syncthing testing stuff in /tmp (`createTmpDir` exists for that purpose).